src/core/etl/src/Flow/ETL/Row/Factory/NativeEntryFactory.php

```diff
@@ -56,64 +75,64 @@ public function create(string $entryName, mixed $value, ?Schema $schema = null)
 
                 if ('' !== $trimmedValue) {
                     if ($this->isJson($trimmedValue)) {
-                        return new Row\Entry\JsonEntry($entryName, $value);
+                        return json_entry($entryName, $value);
```

Comment:
I'm not sure we should use internal calls to DSL functions, as those are internal things not exposed anywhere IMO it could be left as it was.

Maybe mark those classes as internal? This would prevent initialization outside of Flow's internal code.

Reply:
I don't see anything wrong with using DSL internally. 
I would even say that no code should be merged before we make sure that DSL (in case of new features) is defined. Took me a while to go through all types and defined DSL for them.